pkg/networkserver/redis/registry.go

```diff
@@ -76,8 +65,42 @@ func (r *DeviceRegistry) Init(ctx context.Context) error {
 	return nil
 }
 
+// marshalMsgpack is adaptation of msgpack.Marshal (https://github.com/vmihailenco/msgpack/blob/8b121f7d2c8eac21ab98ef6443d272f355b88999/encode.go#L58-L74).
+func marshalMsgpack(v interface{}) ([]byte, error) {
+	enc := msgpack.GetEncoder()
+
+	var buf bytes.Buffer
+	enc.Reset(&buf)
+	enc.UseCompactFloats(true)
+	enc.UseCompactInts(true)
+	enc.UseCustomStructTag("json")
+
+	err := enc.Encode(v)
+	b := buf.Bytes()
+
+	msgpack.PutEncoder(enc)
```

Comment:
Why not `defer` after `GetEncoder` ? I understand that this is close to the original `msgpack.Marshal` but I don't think it hurts to do this.

Reply:
This is copy-paste of original implementation and I would prefer to keep in that way. Why exactly would we want to refactor this into using `defer` except for cosmetic pleasure?

Reply:
`defer` has the advantage of returning the buffer even when a `panic` occurs. But in this particular case I don't expect a panic, so feel free to merge as is (I did approve this).

Reply:
We'd be better off not adding the value back into the pool if this happens and just let it be GC'ed, since the state of `enc` in that case is unknown and it may be dirty